frontend/src/main/scala/bloop/reporter/BspProjectReporter.scala

```diff
@@ -16,22 +16,40 @@ import scala.collection.mutable
 import scala.util.Try
 import scala.concurrent.Promise
 import bloop.CompileOutPaths
+import monix.execution.atomic.AtomicInt
 
 final class BspProjectReporter(
     val project: Project,
     override val logger: BspServerLogger,
     cwd: AbsolutePath,
     config: ReporterConfig,
     reportAllPreviousProblems: Boolean,
-    override val _problems: mutable.Buffer[ProblemPerPhase] = mutable.ArrayBuffer.empty
+    override val _problems: Reporter.Buffer[ProblemPerPhase]
 ) extends Reporter(logger, cwd, config, _problems) {
+
+  def this(
+      project: Project,
+      logger: BspServerLogger,
+      cwd: AbsolutePath,
+      config: ReporterConfig,
+      reportAllPreviousProblems: Boolean
+  ) =
+    this(
+      project,
+      logger,
+      cwd,
+      config,
+      reportAllPreviousProblems,
+      createBuffer[ProblemPerPhase](project)
+    )
+
   private lazy val taskId = logger.nextTaskId
 
   /** A cycle count, initialized to 0 when it's a no-op. */
-  private var cycleCount: Int = 0
+  private val cycleCount: AtomicInt = AtomicInt(0)
```

Comment:
Yes, otherwise there can be visibility problems due to the `Reporter` instance being accessed from different threads. Also, the commit message in https://github.com/scalacenter/bloop/pull/1190/commits/ae26519b437881291880c116740e5d649dcdd63b provides context on this change (in case you missed it)

Reply:
I'm going to mark this resolved.